File: aspnet-in-practice/16/CS/ParallelFX/Code/Providers/FirstProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Threading;

namespace ASPNET4InPractice.Chapter15
{
    public class FirstProvider : IFlightPriceProvider
    {
        public FlightPriceResult GetFlightPrice(string FlightNumber)
        {
            // let's simulate a call to a remote resource
            Thread.Sleep(2000);

            // and return a fixed value
            return new FlightPriceResult() { FlightNumber = FlightNumber, Price = 560 };
        }

    }

}

File: aspnet-in-practice/16/VB/ThreadPool/App_Code/Providers/FirstProvider.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Threading

Public Class FirstProvider
	Implements IFlightPriceProvider
	Public Function GetFlightPrice(ByVal flightNumber As String) As FlightPriceResult Implements IFlightPriceProvider.GetFlightPrice
		' let's simulate a call to a remote resource
		Thread.Sleep(2000)

		' and return a fixed value
		Dim result As New FlightPriceResult()
		result.FlightNumber = flightNumber
		result.Price = 560
		Return result
	End Function

End Class